xml/src/main/java/org/ehcache/xml/CoreCacheConfigurationParser.java

```diff
@@ -0,0 +1,100 @@
+import org.ehcache.expiry.ExpiryPolicy;
+import org.ehcache.xml.model.CacheTemplate;
+import org.ehcache.xml.model.CacheType;
+import org.ehcache.xml.model.Expiry;
+import org.ehcache.xml.model.ExpiryType;
+import org.ehcache.xml.model.TimeType;
+import org.ehcache.xml.model.TimeUnit;
+
+import java.math.BigInteger;
+import java.time.Duration;
+
+import static org.ehcache.core.spi.service.ServiceUtils.findAmongst;
+import static org.ehcache.xml.XmlConfiguration.getInstanceOfName;
+
+public class CoreCacheConfigurationParser {
+
+  public <K, V> CacheConfigurationBuilder<K, V> parseConfiguration(CacheTemplate cacheDefinition, ClassLoader cacheClassLoader,
+                                                                   CacheConfigurationBuilder<K, V> cacheBuilder) throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+    final Expiry parsedExpiry = cacheDefinition.expiry();
+    if (parsedExpiry != null) {
+      cacheBuilder = cacheBuilder.withExpiry(getExpiry(cacheClassLoader, parsedExpiry));
+    }
+
+    EvictionAdvisor evictionAdvisor = getInstanceOfName(cacheDefinition.evictionAdvisor(), cacheClassLoader, EvictionAdvisor.class);
+    cacheBuilder = cacheBuilder.withEvictionAdvisor(evictionAdvisor);
+
+    return cacheBuilder;
+  }
+
+  @SuppressWarnings({"unchecked", "deprecation"})
+  private static ExpiryPolicy<? super Object, ? super Object> getExpiry(ClassLoader cacheClassLoader, Expiry parsedExpiry)
+    throws ClassNotFoundException, InstantiationException, IllegalAccessException {
+    final ExpiryPolicy<? super Object, ? super Object> expiry;
+    if (parsedExpiry.isUserDef()) {
+      ExpiryPolicy<? super Object, ? super Object> tmpExpiry;
+      try {
+        tmpExpiry = getInstanceOfName(parsedExpiry.type(), cacheClassLoader, ExpiryPolicy.class);
+      } catch (ClassCastException e) {
+        tmpExpiry = ExpiryUtils.convertToExpiryPolicy(getInstanceOfName(parsedExpiry.type(), cacheClassLoader, org.ehcache.expiry.Expiry.class));
+      }
+      expiry = tmpExpiry;
+    } else if (parsedExpiry.isTTL()) {
+      expiry = ExpiryPolicyBuilder.timeToLiveExpiration(Duration.of(parsedExpiry.value(), parsedExpiry.unit()));
+    } else if (parsedExpiry.isTTI()) {
+      expiry = ExpiryPolicyBuilder.timeToIdleExpiration(Duration.of(parsedExpiry.value(), parsedExpiry.unit()));
+    } else {
+      expiry = ExpiryPolicyBuilder.noExpiration();
+    }
+    return expiry;
+  }
+
+  public CacheType unparseConfiguration(CacheConfiguration<?, ?> cacheConfiguration, CacheType cacheType) {
+    ExpiryPolicy<?, ?> expiryPolicy = cacheConfiguration.getExpiryPolicy();
+    if (expiryPolicy != null) {
+      ExpiryType expiryType = new ExpiryType();
+      if (expiryPolicy == ExpiryPolicy.NO_EXPIRY) {
+        expiryType.withNone(new ExpiryType.None());
+      } else if (expiryPolicy instanceof ExpiryPolicyBuilder.TimeToLiveExpiryPolicy) {
+        Duration ttl = expiryPolicy.getExpiryForCreation(null, null);
```

Comment:
I think you can pull this out and then use it to generate a test instance through the builder and check for equality of the implementation class.  In essence we don't care *what* the impl class is, just that it matches this one.

Reply:
Will do